### PR TITLE
Add support for filtering by changelog type for testing instructions

### PR DESCRIPTION
**Pull request by louwie17**
Add support for filtering by changelog type for testing instructions as we usually don't want all of them.
There was also a couple PHP warnings that shouldn't show up anymore, and I updated the formatting a bit, so it shouldn't be adding extra new lines to the testing instructions file.

### Detailed test instructions:

- Make sure you have the Git credentials set up by running: `npm run test-instruction-logger github-credentials`
- Run `npm run test-instruction-logger -- write 3.2.0 --types=enhancement,add`
- Check TESTING-INSTRUCTIONS.md and see if only the changes for types enhancements and add are added. (you can try other types as well, like `fix`).
- Check the formatting to see if things look good.

### CODE REVIEW

File: bin/changelogger/WCAdminFormatter.php [hunk 1]
```diff
@@ -103,12 +103,9 @@ function ( $m ) {
...
 
-					array_push(
-						$changes,
-						array(
-							'subheading' => trim( $row_segments[0] ),
-							'content'    => trim( $row_segments[1] ),
-						)
+					$changes[] = array(
+						'subheading' => trim($row_segments[0]),
+						'content' => count( $row_segments ) > 1 ? trim($row_segments[1]) : '',
```
Comment (louwie17) [inline bin/changelogger/WCAdminFormatter.php:108]: row_segments only had one item sometimes, which caused a warning, this prevents that.

File: bin/test-instruction-logger/WriteCommand.php [hunk 2]
```diff
@@ -137,30 +143,28 @@ protected function buildTestInstructions( $testingInstructionsPath, $prContents,
 		$output = array();
 
 		// Get the first line (heading) from the TESTING-INSTRUCTIONS.md.
-		array_push( $output, array_shift( $testingInstructions ) );
+		$output[] = array_shift( $testingInstructions );
```
Comment (louwie17) [inline bin/test-instruction-logger/WriteCommand.php:146]: phpStorm tells me that this way is 2 times faster then `array_push`.

**Review by moon0326 — APPROVED** [review]
Thank you for working on this @louwie17! Nice feature 🎉 

It LGTM and tested well. 

I never wrote a P2 about it because we're in the middle of the merge and I wasn't sure how this logger was going to be used, but it's nice that you're using it :)

**Comment by louwie17** [comment]
> I never wrote a P2 about it because we're in the middle of the merge and I wasn't sure how this logger was going to be used, but it's nice that you're using it :)

Yeah it worked pretty well aside from some of the formatting issues and not being able to filter out which PRs to grab it from. Much nicer this way, then having to load them all up ourselves and copy/paste.

**Pull request merged**